sigsci_site_manager/user.py

```diff
@@ -0,0 +1,165 @@
+        if args.site_name:
+            print('Site: %s' % args.site_name)
+            print(colFormat % (cols[0], 'site-' + cols[1],
+                               cols[2], cols[3]))
+            for user in users['data']:
+                line_entries.append(colFormat % (user['user'][cols[0]], user[cols[1]],
+                                                 user['user'][cols[2]], user['user'][cols[3]]))
+        else:
+            print(colFormat % (cols[0], 'corp' + cols[1],
+                               cols[2], cols[3]))
+            for user in users['data']:
+                line_entries.append(colFormat % (user[cols[0]], user[cols[1]],
+                                                 user[cols[2]], user[cols[3]]))
+        print_sorted_array(line_entries)
+
+
+def do_list_membership(args):
+    api = init_api(args.username, args.password, args.token, args.corp,
+                   args.dry_run)
+    if args.email_id:
+        memberships = api.get_memberships(args.email_id)
+        cols = ['role', 'site']
+        colFormat = "%10s %s"
+        if memberships:
+            line_entries = []
+            for member in memberships['data']:
+                line_entries.append(colFormat % (member[cols[0]],
+                                                 "%s [%s]" % (member[cols[1]]['displayName'],
+                                                              member[cols[1]]['name'])))
+            if len(line_entries) > 0:
+                print(colFormat % (cols[0], cols[1]))
+                print_sorted_array(line_entries)
+
+
+def do_add_user(args):
+    api = init_api(args.username, args.password, args.token, args.corp,
+                   args.dry_run)
+
+    input_file = resolve_input_file(args.file_name)
+    location_string = 'corp'
+    if args.site_name:
+        api.site = args.site_name
+        location_string = args.site_name
+
+    user_list = []
+    if args.email_id:
+        user_list.append([args.email_id, args.role_name])
+    if input_file:
+        user_list = get_users_from_file(input_file, 2)
+
+    added_users = 0
+    for entry in user_list:
+        user_role = entry[1]
+        if args.role_name and len(entry[1]) >= 0:
+            user_role = args.role_name
+        data = resolve_user_data_role(api, user_role)
+        if args.dry_run:
+            print("Adding %s to %s with %s role" %
+                  (entry[0], location_string, data['role']))
+        add_new_user(api, entry[0], data['role'], args.api_user)
+        added_users += 1
+    print("Added %d users" % added_users)
+
+
+def do_remove_user(args):
+    api = init_api(args.username, args.password, args.token, args.corp,
+                   args.dry_run)
+
+    input_file = resolve_input_file(args.file_name)
+    location_string = 'corp'
+    if args.site_name:
+        api.site = args.site_name
+        location_string = args.site_name
+
+    user_list = []
+    if args.email_id:
+        user_list.append([args.email_id])
```

Comment:
Did you mean to append a list to a list? This results in `user_list = [[args.email.id]]`, not `user_list = [args.email]`. Is that what is intended?

Reply:
yes, this is intentional, enables the use of either one or more column entries depending on the data in the parsed file